dpgen/generator/lib/calypso.py

```diff
@@ -0,0 +1,438 @@
+    ret+="    for i in range(3):                                                                  \n"
+    ret+="        f.write('%15.10f %15.10f %15.10f\\n' % tuple(lat[i]))                           \n"
+    ret+="    for x in element:                                                                   \n"
+    ret+="        f.write(x + '  ')                                                               \n"
+    ret+="    f.write('\\n')                                                                      \n"
+    ret+="    for x in element:                                                                   \n"
+    ret+="        f.write(str(ele[x]) + '  ')                                                     \n"
+    ret+="    f.write('\\n')                                                                      \n"
+    ret+="    f.write('Direct\\n')                                                                \n"
+    ret+="    na = sum(ele.values())                                                              \n"
+    ret+="    dpos = np.dot(pos,np.linalg.inv(lat))                                               \n"
+    ret+="    for i in range(na):                                                                 \n"
+    ret+="        f.write('%15.10f %15.10f %15.10f\\n' % tuple(dpos[i]))                          \n"
+    ret+="                                                                                        \n"
+    ret+="def Write_Outcar(element, ele, volume, lat, pos, ene, force, stress,pstress):           \n"
+    ret+="    '''Write OUTCAR'''                                                                  \n"
+    ret+="    f = open('OUTCAR','w')                                                              \n"
+    ret+="    for x in element:                                                                   \n"
+    ret+="        f.write('VRHFIN =' + str(x) + '\\n')                                            \n"
+    ret+="    f.write('ions per type =')                                                          \n"
+    ret+="    for x in element:                                                                   \n"
+    ret+="        f.write('%5d' % ele[x])                                                         \n"
+    ret+="    #f.write('\\nvolume of cell :\\n')                                                  \n"
+    ret+="    f.write('\\nDirection     XX             YY             ZZ             XY             YZ             ZX\\n') \n"
+    ret+="    f.write('in kB')                                                                    \n"
+    ret+="    f.write('%15.6f' % stress[0])                                                       \n"
+    ret+="    f.write('%15.6f' % stress[1])                                                       \n"
+    ret+="    f.write('%15.6f' % stress[2])                                                       \n"
```

Comment:
I strongly suggest put these codes in  a complete script, like `lib/xxx.py`, instead of making them strings like you did. On the one hand, the code is easier to read and maintain. On the other hand, users can directly run the script if they have needs. 
If you need to execute the script, you can copy the script to you current path. 
You can refer to
https://github.com/deepmodeling/dpgen/blob/93b833ae3602d2e9908e7e4c6b639a3b381d4c15/dpgen/generator/run.py#L1911